### PR TITLE
Update templates, issues and PR's

### LINKED ISSUE #824: Add a pull request template
This should just be a quick checklist asking people to run the check script, maybe mention that review could take a week etc?

### PR DESCRIPTION
**Pull request by LiHRaM**
This change moves the files into the `.github/` folder, and adds a new pull request template.

closes #824

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Pull Request Title
+Quickly summarize the purpose of your PR.
+
+## Related issues
+Give reviewers and interested parties a good idea of how this is related to other issues or pull requests.
+
+Example: `Related to #1`
+
+## Testing
+Describe how you plan to test, or how your changes have been tested.
```
Comment (cmyr) [inline .github/PULL_REQUEST_TEMPLATE.md]: I would mention that you can run all the tests locally by running `./rust/run_all_checks` from repository's base directory.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Xi-Editor Issue Template
+- [ ] I have searched existing issues and could not find my issue.
+- [ ] I have studied the documentation.
+- [ ] This is not a personal support question, which belongs on #xi (irc.mozilla.org)
```
Comment (cmyr) [inline .github/ISSUE_TEMPLATE.md]: I think 'personal support' is a bit vague. If you can't get xi to compile, I think opening an issue is a reasonable step. Encouraging people to go to IRC is good, but I'm not sure it should be mandatory?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (LiHRaM) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'll add that, yeah. I've also edited the document with comments - feels safer when you don't have to worry about them accidentally posting the entire template in the PR. 😄

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (LiHRaM) [reply .github/ISSUE_TEMPLATE.md]: That's 👍 , I had just looked at other examples that said something like this, but I have no strong feelings on the subject.

**Pull request merged**